File: php7/src/ChainOfResponsibility/BankCustomer.php
<?php

declare(strict_types = 1);

namespace PetrePatrasc\ChainOfResponsibility;

use PetrePatrasc\ChainOfResponsibility\Bank\LoanSystem;
use PetrePatrasc\ChainOfResponsibility\Bank\LoanSystemBuilder;

/**
 * A representation of the bank customer that wants to apply for a loan.
 *
 * @package PetrePatrasc\ChainOfResponsibility
 * @author  Petre Pătraşc <petre@dreamlabs.ro>
 */
class BankCustomer implements BankCustomerInterface
{
    /**
     * @var string
     */
    protected $name;

    public function __construct($name)
    {
        $this->setName($name);
    }

    /**
     * @inheritDoc
     */
    public function applyForLoan(float $amount, LoanSystem $loanSystem): LoanApplicationResponse
    {
        $loanRequest = new LoanApplicationRequest($this, $amount);

        return $loanSystem->resolveLoanRequest($loanRequest);
    }

    /**
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param string $name
     *
     * @return BankCustomer
     */
    public function setName(string $name): BankCustomer
    {
        $this->name = $name;

        return $this;
    }
}

File: java/src/ro/petrepatrasc/chainofresponsability/customer/BankCustomer.java
package ro.petrepatrasc.chainofresponsability.customer;

import ro.petrepatrasc.chainofresponsability.bank.LoanSystem;
import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationRequest;
import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationResponse;

/**
 * A representation of the bank customer that wants to apply for a loan.
 *
 * @author Petre Pătrașc <petre@dreamlabs.ro>
 */
public class BankCustomer implements BankCustomerInterface {

    protected String name;

    public BankCustomer(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return this.getName();
    }

    @Override
    public LoanApplicationResponse applyForLoan(double amount, LoanSystem loanSystem) {
        LoanApplicationRequest loanRequest = new LoanApplicationRequest(this, amount);

        return loanSystem.resolveLoanRequest(loanRequest);
    }

    public String getName() {
        return name;
    }

    public BankCustomer setName(String name) {
        this.name = name;
        return this;
    }
}
